packages/babel-parser/src/plugins/typescript/index.js

```diff
@@ -2930,6 +2932,17 @@ export default (superClass: Class<Parser>): Class<Parser> =>
         return expr;
       }, state);
 
+      // report error if single type parameter used without trailing comma.
+      if (
+        arrow.node?.type === "ArrowFunctionExpression" &&
+        arrow.node?.typeParameters?.params.length === 1 &&
+        !arrow.node?.typeParameters?.extra?.trailingComma
+      ) {
+        const typeName = arrow.node.typeParameters.params[0].name;
+        const start = arrow.node.start;
+        this.raise(start, TSErrors.SingleTypeWithoutTrailingComma, typeName);
```

Comment:
The Babel 8 tests (you can run them with `BABEL_8_BREAKING=true yarn jest parser`) are failing, I think it's becaus `typeName` isn't a string anymore but a node: https://github.com/babel/babel/pull/12829